h2/src/tools/org/h2/build/BuildBase.java

```diff
@@ -962,6 +962,24 @@ protected void javac(StringList args, FileList files) {
         println("Compiling " + files.size() + " classes");
         StringList params = new StringList();
         params.addAll(args);
+        
+        // Bugfix - Use UTF-8 instead of the default locale encoding when compiling,
+        //because of H2 source file encoding is UTF-8.
+        // @since 2018-07-19 little-pan
+        String encoding = null;
+        for(int i = 0, size = params.size(); i < size; ++i){
+        	if("-encoding".equals(params.get(i))){
+        		encoding = params.get(i + 1);
+        		break;
+        	}
+        }
+        if(encoding == null){
+        	encoding = StandardCharsets.UTF_8.name();
+        	params.add("-encoding");
+        	params.add(encoding);
+        }
+        println("Using encoding " + encoding);
```

Comment:
Do you really need to see name of the encoding in build output? Seems to be useless.

All characters in H2 sources are from ASCII, and even if it will be changed it's very unlikely that something different from UTF-8 will be used.